app/services/media_storage/stored_path.rb

```diff
@@ -10,11 +10,14 @@ def self.media_url(url, stored_path)
       PublicSuffix.parse(uri.host)
       # we have a valid domain prefix here so remove it
       # to allow us to construct the URL correctly
+      env_prefix = '/' + Rails.env
+      uri.path.slice! env_prefix if uri.path.start_with? env_prefix # remove env prefix if present
```

Comment:
I prefer a regex with capture group for this use case rather than testing if the path starts with and if so slicing the substring out. A regex could handle a missing leading `/` too.

I think the code is simpler this way but i'll defer to you, e.g. regex like 
`\A(?:\/?#{Rails.env})?(.+)\z`

where `\A` is start of string, `\z` is end of string
`(?:\/?production)?` is a non-capture group `(?:)`, it may have a leading `/` (zero or more) followed by the rails env we want to ignore, note this non-capture group doesn't have to be present but will match if it is.
`(.+)` the greedy capture group for the resource path we want to serve
try it at https://rubular.com/ with input like
```
/user_avatar/1e5fc9b5-86f1-4df3-986f-549f02f969a5.jpeg
production/user_avatar/1e5fc9b5-86f1-4df3-986f-549f02f969a5.jpeg
/test/user_avatar/1e5fc9b5-86f1-4df3-986f-549f02f969a5.jpeg
/staging//user_avatar/1e5fc9b5-86f1-4df3-986f-549f02f969a5.jpeg
```
e.g. 
```
# micro op make this regex a class constant or class method to avoid instantiating it each time this method is called
RAILS_ENV_REMOVAL_REGEX = /\A(?:\/?production)?(.+)\z/

# get the 1st capture group (0 index is the original string) which should be the trailing media type / UUID regex etc.
uri_path = uri.match(RAILS_ENV_REMOVAL_REGEX)[1]
```